Msband

VRocker opened the pull request:
### Changes:

- Added Microsoft Band integration
- Added the ability to send notifications of nearby pokemon to the Microsoft Band
- Added a new option to the settings page for Band

### Change details:

- Implemented the Band SDK and created a new helper for band stuff

### Other informations:
Currently i've only been able to test using the Band 1 but the SDK should work for both perfectly fine. Notification messages only work if the tile has been added to the band which requires a free tile slot (Bands can only hold 13 tiles at a time). The app will tell you if there's no free slots.
Even if the tile hasn't been added, the band will still vibrate to tell you that there's a nearby pokemon.

I realise it will be difficult for ST-Apps to test this implementation if he doesn't have a Band so i guess it's up to the rest of us which do have them to test and confirm that this works.

VRocker commented:
Reopened pull request to fix conflicts. I had merged another branch into this one for testing purposes.
I've removed that merge and reopened the PR.

Maxyeah commented:
OMG all my dreams come true!!! You are so awesome. Can't wait to test the feature with my band 2. Is it already possible to test the feature if I compile the app by myself? So I could tell u how it works with band2 if needed. Yeah an it would be cool if it is possible to get miles for the eggs just with wearing the band, because when I go running I don't carry my phone with me :). So if I can help testing this feature just tell me.

VRocker commented:
Yeah if you pull down the msband branch from my fork you should be able to compile it yourself.

As for eggs, that's a great suggestion that would be a really neat feature. Not actually sure how the egg hatching works yet though so i'd need to look into if it's possible.

Tweeter1 commented:
Watching...😀

Chatfix commented:
@ST-Apps please be careful with merging this PR. 
I've testet VRockers fork and he helped me to get it work, but I think there's a little more work with Google.Protobufs conflicts.

For now: I didn't get it work.
I try again later.

VRocker commented:
I've created a new branch which fixes the protobufs issue. I didn't think the protobufs would cause an issue with the band as i was experiencing the issue before adding the band SDK.

Either way, could some try this branch? https://github.com/VRocker/PoGo-UWP/tree/msband%2Bprotobuf

Tweeter1 commented:
Error when building app package (VS2015)
Severity	Code	Description	Project	File	Line	Suppression State
Error		Payload contains two or more files with the same destination path 'Google.Protobuf.dll'. Source files: 
C:\Users\flopp_000\.nuget\packages\Google.Protobuf\3.0.0\lib\netstandard1.0\Google.Protobuf.dll
C:\Users\flopp_000\.nuget\packages\Microsoft.Band\1.3.20628\lib\uap10.0\Google.Protobuf.dll	PokemonGo-UWP

VRocker commented:
Hmm chatfix had the same issue (and i did a while ago too). Funny thing is that i don't even have the google.protobuf.dll file in that directory as i deleted it. The Band SDK doesn't even use protobufs so i don't know why it gets put in there.
Could try deleting the band folder and restoring the nuget package?

Tweeter1 commented:
No luck with deleting .dll or removing/adding MSband NuGet.

Forceh91 commented:
I've just done a fresh pull of this branch and it compiled fine for me.

Chatfix commented:
Tried your linked branch, but same error: 

> Error Payload contains two or more files with the same destination path 'Google.Protobuf.dll'. Source files: 
 C:\Users\flopp_000.nuget\packages\Google.Protobuf\3.0.0\lib\netstandard1.0\Google.Protobuf.dll
 C:\Users\flopp_000.nuget\packages\Microsoft.Band\1.3.20628\lib\uap10.0\Google.Protobuf.dll

Tweeter1 commented:
LOL. was just reading that one.  basically just pick one and copy paste so they are the same version.

christosk92 commented:
@Chatfix yes

christosk92 commented:
@Tweeter1 Yes.

tofylion commented:
Will this be added in the next update?

Tweeter1 commented:
Does not appear to work on my MS Band 2.  It did add the Tile to my band.

leatherrabbit commented:
Like a dream come true. I will wait until the feature is fully integrated into the PoGo app to really make use of it, but compiled and works fine. Band 1. Can't comment on Band 2. Will try it tomorrow or the day after, when I have access to one.

VRocker commented:
Fixed a bug that stopped the messages from being shown on the band and also fixed the conflicts.

Pic from my earlier walk: http://imgur.com/4dB29J1

Maxyeah commented:
Ok i tried to compile it but still get the same Error as Chatfix and i don't know how to handle the solution of christosk92. 
Deploying the msband (without proto) branch will show this error .

> Fehler CS0433	
Der Typ "RepeatedField<T>" ist in "Google.Protobuf, Version=3.0.0.0, Culture=neutral, PublicKeyToken=a7d26565bac4d604" und "Google.Protobuf, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" vorhanden.	
PokemonGo-UWP	C:\Users\user\Desktop\PokemonGo\PoGo-UWP-msband\PoGo-UWP-msband\PokemonGo-UWP\Entities\FortDataWrapper.cs

Tweeter1 commented:
@Maxyeah copy the dll from the google folder over to the msband folder. ..read the post above.

chuckfecht commented:
I tried this on MS Band 2, it works!! Now lets just get the distance traveled to count towards egg hatching :D

Maxyeah commented:
Ok, I was able to deploy it but there were too many error messages and i don't get any notifications on my band 2. 
![wp_ss_20160824_0001](https://cloud.githubusercontent.com/assets/16652433/18030723/c940ca10-6cc1-11e6-8ae4-e5ac33e60e9a.png)

chuckfecht commented:
@Maxyeah You need to copy and replace the **Google.Protobuf.dll** from 
`C:\Users\[youruser]\.nuget\packages\Google.Protobuf\3.0.0\lib\netstandard1.0\Google.Protobuf.dll`
 to 
`C:\Users\[youruser]\.nuget\packages\Microsoft.Band\1.3.20628\lib\uap10.0\Google.Protobuf.dll`
It looks like that's your issue, you'll need to rebuild and redeploy to get it working properly

beneathwithme commented:
Does the current PoGo API support Pokémon Go Plus functionality? I was thinking we could use the band as a substitute for that, and use it to catch Pokémon, and make it vibrate when a Pokemon appears, say what it is, etc.

Jehu147 commented:
How can I install it in my band please i want to use like it would be the pokemon go plus I am so excited

zakius commented:
I think with pogo coming (partially) to Apple Watch this one needs more attention

magic4got10 commented:
Is there a way to get a vibration on my MiBand 1S? On Windows I use the clint Bind Mi Band, that has recently added notification's support with a beta app called Notify Mi Band that works great. So would it be hard to send the same vibration that I feel on the phone, on my Mi Band?
